### PR TITLE
main/ncurses: update to 6.5

### PR DESCRIPTION
**Pull request by nekopsykose**

### CODE REVIEW

File: main/ncurses/template.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 pkgname = "ncurses"
-pkgver = "6.4"
-pkgrel = 2
+pkgver = "6.5"
+pkgrel = 0
 build_style = "gnu_configure"
 configure_args = [
+    "--disable-root-access",
```
Comment (nekopsykose) [inline main/ncurses/template.py:6]: there's also root-environ but idk if that should also be disabled

Comment (q66) [reply main/ncurses/template.py:6]: all i can find is that arch enables it and then https://github.com/kovidgoyal/kitty/issues/6842

they still enable it https://gitlab.archlinux.org/archlinux/packaging/packages/ncurses/-/blob/main/PKGBUILD?ref_type=heads#L55

**Comment by q66** [comment]
i also wonder if `--enable-fvisibility` could be used... or if it causes some stuff to break

File: main/ncurses/template.py (hunk 1 above)
Comment (nekopsykose) [reply main/ncurses/template.py:6]: better to keep it on then :)

**Comment by nekopsykose** [comment]
> i also wonder if --enable-fvisibility could be used... or if it causes some stuff to break  

afaict it does the same thing hardening=vis would do so i would set that instead. since it's now upstream it probably doesn't break anything...

**Comment by q66** [comment]
if using vis we'd still need the configure arg for robustness since the build system can take different choices based on the setting and ncurses build system is not exactly known for not doing sketchy things

**Comment by nekopsykose** [comment]
sure

i checked symbol diff and is the same

**Comment by q66** [comment]
nice

File: main/ncurses/template.py (hunk 1 above)
Comment (q66) [reply main/ncurses/template.py:6]: debian disables root-access but not root-environ

Comment (nekopsykose) [reply main/ncurses/template.py:6]: guess we'll match that then

**Pull request merged**